### PR TITLE
feat: error resilient errors

### PR DESCRIPTION
**Pull request by parth121101**
Towards: https://github.com/lfortran/lfortran/issues/5036

### CODE REVIEW

**Comment by parth121101** [comment]
Will add more errors in this PR itself.
Till that if someone can review, it will be helpful

**Comment by certik** [comment]
Looks good. Get CI to pass and let's merge it.

**Review by Pranavchiku — DISMISSED** [review]
update test references using:

```
rm -rf ./tests/reference/*
./run_tests.py -u
```

**Comment by parth121101** [comment]
> update test references using:
> 
> ```
> rm -rf ./tests/reference/*
> ./run_tests.py -u
> ```

aye aye, sir

File: tests/errors/continue_compilation_2.f90 [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+
+    ! cmplx_02
+    print*, cmplx((real(1, kind=4), 0.00000000), kind=8)
+    print *, "Seventh error skipped"
+
+
+    ! cmplx_03
+    print*, cmplx((1.00000000, real(0, kind=4)), kind=8)
+    print *, "Eighth error skipped"
```
Comment (certik) [inline tests/errors/continue_compilation_2.f90:78]: Let's shorten these to just:
```fortran
    CLOSE(end=200)
    a = cmplx(y = 2)
    print*, cmplx((real(1, kind=4), 0.00000000), kind=8)
    print*, cmplx((1.00000000, real(0, kind=4)), kind=8)
```
Seems a lot shorter and removes noise.

**Pull request merged**